.github/workflows/gen-check.yaml

```diff
@@ -0,0 +1,19 @@
+# Copyright The OpenTelemetry Authors
+# SPDX-License-Identifier: Apache-2.0
+
+name: Clean Generation
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+jobs:
+  protobufcheck:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Generate
+        run: make clean docker-generate-protobuf
+      - name: Check Clean Work Tree
+        run: make check-clean-work-tree
```

Comment:
@rogercoll how this would change the current workflow?
This new check would fail if the protobuf is outdated, no!?

Reply:
It would be part of the build workflow, a new job that would be needed before building the images. Similar to the `prepare` job requirement in https://github.com/open-telemetry/opentelemetry-collector-releases/blob/main/.github/workflows/base-release.yaml#L110

Reply:
A ha!
That makes sense, and if it fails, we not even build the images.
I like this!